gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/activity/impl/CommitActivityImpl.java

```diff
@@ -135,6 +137,12 @@ public Callable<Void> apply(final Map.Entry<String, JobState.DatasetState> entry
 
       IteratorExecutor.logFailures(result, null, 10);
 
+      for (JobState.DatasetState datasetState : datasetStatesByUrns.values()) {
+        // Set the overall job state to FAILED if the job failed to process any dataset
+        if (datasetState.getState() == JobState.RunningState.FAILED) {
+          throw new IOException("Failed to commit dataset state for dataset " + datasetState.getDatasetUrn());
```

Comment:
Do we want to log / throw an exception that represents all the failures? As opposed to just 1?